stdlib/list.ml

```diff
@@ -197,16 +197,26 @@ let partition p l =
   | x :: l -> if p x then part (x :: yes) no l else part yes (x :: no) l in
   part [] [] l
 
-let rec split = function
-    [] -> ([], [])
-  | (x,y)::l ->
-      let (rx, ry) = split l in (x::rx, y::ry)
+let rev_split xys =
+  let rec _rev_split xs ys = function
```

Comment:
Using identifiers starting with `_` for an internal helper function is not idiomatic (this prefix is used to disable the "unused value" warning).  `rev_split_aux` or just `aux` would be better.